test/comprehensive/host/testdeepcopy.cpp

```diff
@@ -377,6 +377,59 @@ void test_deepcopy_edl_ecalls(oe_enclave_t* enclave)
         OE_TEST(s.size == 200);
     }
 
+    {
+        printf("deepcopy_countparam_out started\n");
+        CountParamStruct s;
+        OE_TEST(deepcopy_countparam_out(enclave, &s) == OE_OK);
+        test_struct(s, 5);
+        OE_TEST(s.count == 5);
+        OE_TEST(s.size == 200);
+        printf("deepcopy_countparam_out passed\n");
+    }
+
+    {
+        SizeParamStruct s;
+        OE_TEST(deepcopy_sizeparam_out(enclave, &s) == OE_OK);
+        test_struct(s, 5);
+        OE_TEST(s.count == 100);
+        OE_TEST(s.size == 5 * sizeof(uint64_t));
+        printf("deepcopy_sizeparam_out passed\n");
+    }
+
+    {
+        CountSizeParamStruct s;
+        OE_TEST(deepcopy_countsizeparam_out(enclave, &s) == OE_OK);
+        test_struct(s, 5);
+        OE_TEST(s.count == 5);
+        OE_TEST(s.size == sizeof(uint64_t));
+        printf("deepcopy_countsizeparam_out passed\n");
+    }
+
+    {
+        CountSizeStruct s;
+        OE_TEST(deepcopy_countsize_out(enclave, &s) == OE_OK);
+        test_struct(s, 3);
+        OE_TEST(s.count == 100);
+        OE_TEST(s.size == 200);
+        printf("deepcopy_countsize_out passed\n");
+    }
+
+    {
+        NestedStruct n;
+        OE_TEST(deepcopy_nested_out(enclave, &n) == OE_OK);
+        OE_TEST(n.plain_int == 100);
+        for (int i = 0; i < 4; i++)
+            OE_TEST(n.array_of_int[i] == i);
+        OE_TEST(n.shallow_struct == NULL);
```

Comment:
Where are array_of_int, shallow_struct and array_of_struct properties freed?

Reply:
Yes, it's currently missed in this PR. In this design, both callee and caller would need to free the nested buffers.

Reply:
The has been taken care by the current implementation.